spec/rails_shim_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe 'Apivore::RailsShim' do
+
+  describe 'interpreting ActionDispatch request arguments' do
+    subject {
+      Apivore::RailsShim.action_dispatch_request_args(
+        path,
+        params: params,
+        headers: headers
+      )
+    }
+    let(:path) { '/posts' }
+    let(:params) { { 'foo' => 'bar' } }
+    let(:headers) { { 'X-Foo' => 'baz' } }
+
+    before do
+      stub_const('ActionPack::VERSION::MAJOR', actionpack_major_version)
+    end
+
+    context 'using Rails 4' do
+      let(:actionpack_major_version) { 4 }
+
+      it 'yields data and headers as 2 hashes, in Rails 4 format' do
+        expect(subject).to eq([path, params, headers])
+      end
+    end
+
+    context 'using Rails 5' do
+      let(:actionpack_major_version) { 5 }
+
+      it 'yields data and headers in 1 hash, using Rails 5 kwargs' do
```

Comment:
`'returns path as a positional argument and params and headers as keyword arguments'`